### PR TITLE
test failure notify

### PR DESCRIPTION
**Pull request by Bento007**
#### Reviewers
**Functional:** @MDunitz 

**Readability:** @tihuan 

---

## Changes
- add slack notification when main fails
- add slack notification when staging and prod deploy

### CODE REVIEW

File: backend/corpora/common/authorizer.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def assert_authorized_token(token: str) -> dict:
     auth0_domain = auth_config.internal_url
     audience = auth_config.audience
     public_keys = get_public_keys(auth0_domain)
-    public_key = public_keys.get(unverified_header["kid"])
+    public_key = public_keys.get(unverified_header['kid'])
```
Comment (Bento007) [inline backend/corpora/common/authorizer.py:26]:
```suggestion
    public_key = public_keys.get(unverified_header["kid"])
```

File: .github/workflows/push-tests.yml [hunk 2]
```diff
@@ -38,7 +38,17 @@ jobs:
           docker-compose run --no-deps --rm backend make -C /corpora-data-portal lint
       - name: Lint frontend
         run: |
+          npm run prettier-check
+          npm run lint
```
Comment (Bento007) [inline .github/workflows/push-tests.yml:42]:
```suggestion
```

File: .github/workflows/push-tests.yml [hunk 3]
```diff
@@ -39,6 +39,14 @@ jobs:
       - name: Lint frontend
         run: |
           docker-compose run --no-deps --rm frontend make lint
+      - uses: 8398a7/action-slack@v3
```
Comment (maniarathi) [inline .github/workflows/push-tests.yml:42]: Oooh this one is going to be super spammy no? It runs on every push to every non-merged PR in addition to main?

**Review by MDunitz — APPROVED** [review]
lgtm

**Review by tihuan — APPROVED** [review]
Thanks so much for adding this, Trent! This is gonna help a lot 🚀 !

For my own education, how do we specify which Slack channel to send the message to? Thank you!

**Comment by Bento007** [comment]
@tihuan the channel  is in the SLACK_WEBHOOK

**Comment by tihuan** [comment]
Ooh got it! Thanks @Bento007 !!

**Pull request merged**